### ISSUE TITLE
HTTP 403 When Requesting Bandcamp

### ISSUE THREAD

**Issue by yoshiyoshyosh** [7x11x13/free-bandcamp-downloader#11]
Not sure when this started, since I haven't used this program for a while

```
$ ~/.local/python-venv/bin/bcdl-free -l https://lacheque.bandcamp.com -z 12345 -e auto -f FLAC --no-unzip -d lacheque.bandcamp.com/files/
INFO:httpx:HTTP Request: GET https://www.1secmail.com/api/v1/?action=getDomainList "HTTP/1.1 200 OK"
Traceback (most recent call last):
  File "/home/yosh/.local/python-venv/bin/bcdl-free", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/yosh/.local/python-venv/lib/python3.12/site-packages/free_bandcamp_downloader/__main__.py", line 513, in main
    downloader.download_label(arguments["-l"], arguments["--force"])
  File "/home/yosh/.local/python-venv/lib/python3.12/site-packages/free_bandcamp_downloader/__main__.py", line 367, in download_label
    r.raise_for_status()
  File "/home/yosh/.local/python-venv/lib/python3.12/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://lacheque.bandcamp.com/
```

line 367 is
```py
    def download_label(self, url: str, force: bool = False):
        r = self.session.get(url)
        r.raise_for_status()
```

which is odd, since all that `self.session` is is... `self.session = requests.Session()`

...and if I make the request from my own python instance...

```py
Python 3.12.4 (main, Jun  9 2024, 22:05:49) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
 >>> import requests
 >>> s = requests.Session()
 >>> r = s.get('https://lacheque.bandcamp.com')
 >>> r.raise_for_status()
 >>> r
<Response [200]>
```

what's going on here?

**Comment by 7x11x13** [comment]
I'm not able to reproduce... does this happen with all URLs, or just that one?

**Comment by yoshiyoshyosh** [comment]
> does this happen with all URLs, or just that one?

all urls. I thought it was my vpn at first, so I disabled it and tried again--no dice. again, *extremely* odd that it's *only* happening with free-bandcamp-downloader, and not a local python instance

**Comment by yoshiyoshyosh** [comment]
It was urllib3. downgrading urllib3 from 2.2 to 1.26.x made it work again

related issue: https://github.com/urllib3/urllib3/issues/3439

closing since this isn't an issue with free-bandcamp-downloader

**Comment by yoshiyoshyosh** [comment]
actually, this might be able to be worked around in free-bandcamp-downloader with a `requirements.txt` to target <2

**Comment by 7x11x13** [comment]
I'd rather not pin an old urllib3 dependency for all platforms since this bug only affects people running NixOS (?), but we can add a note in the README telling NixOS users that they should install the old version of urllib3.

**Comment by yoshiyoshyosh** [comment]
I'm able to reproduce it on my distribution (void linux). is it not reproducible for you with urllib 2.2.x?

**Comment by 7x11x13** [comment]
I'm not able to reproduce on Armbian with urllib3 2.2.2

**Comment by 7x11x13** [comment]
But I am able to reproduce on Mac...

**Comment by 7x11x13** [comment]
Fixed in v0.3.4

**Comment by pquentin** [comment]
Hello! urllib3 developer here. Bandcamp.com weirdly requires the TLS cipher suites that are used in urllib3 1.26.x, even if it does not use them. Here's the workaround to continue using urllib3 2.x: https://github.com/urllib3/urllib3/issues/3439#issuecomment-2306400349
